lib/utils/token/token.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package token
+
+import (
+	"math"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+// Verify ensures the token fits into our length requirements and
+// its bits of entropy are sufficient.
+func Verify(token []byte) error {
+	return verify(token, defaults.MaxTokenLength)
+}
+
+// VerifyHashed ensures the token fits into our length requirements and
+// its bits of entropy are sufficient. If the token is not going to be
+// hashed by bcrypt before it will be used, use [Verify] instead.
+func VerifyHashed(token []byte) error {
+	return verify(token, defaults.MaxHashedTokenLength)
+}
+
+func verify(token []byte, maxLen int) error {
+	if len(token) < defaults.MinTokenLength {
+		return trace.BadParameter("token is too short, min length is %d", defaults.MinTokenLength)
+	}
+	if len(token) > maxLen {
+		return trace.BadParameter("token is too long, max length is %d", maxLen)
+	}
+
+	entropyBits := TokenStrength(token)
+	if entropyBits < defaults.MinTokenStrength {
+		return trace.BadParameter("token is not strong enough; try with a longer and/or more random token")
```

Comment:
After thinking about this, I'm not sure about this approach from the UX standpoint. This is very vague message and it leaves the user guessing what kind of token is considered strong enough and what does "longer" and "more random" mean. And if a user needs to use a specific tool to generate a proper token, then why not just let Teleport generate it automatically?

Since this originated for the Okta SCIM plugin installation and is used purely for diagnostics/development, I think we should keep things simple. I propose we just hide the --token flag for that command, add a min length check and be done with it. @smallinsky @tcsc Would you agree?

Reply:
Agree we don't need to check entropy and can hide the flag to keep things simple. (For instance user password validation is only limited to length check)



But I feel that this library will be useful for instance we can add a validation check for  where token is unsafe (`tctl tokens add --type=proxy,db --value=123456` ) and this library is not only limited to `Okta SCIM plugin` case.

Reply:
I agree that checking the strength of all user provided tokens would be ideal. Ideally Teleport would always generate necessary tokens but that's not possible with backwards compatibility.

I'm fine with doing a simple length check for Okta SCIM tokens but I don't think it'd be a bad idea to check token strength elsewhere.

Reply:
Let's avoid scope creep. If we all agree that doing a simple min length check for Okta token is good for now, let's do that and re-eval if more sophisticated checks for other tokens are needed once it becomes a priority.

Reply:
In that case I'll make this a draft PR in case we want to add token strength validation elsewhere